executor/infoschema_reader.go

```diff
@@ -2615,54 +2613,77 @@ type tiflashInstanceInfo struct {
 }
 
 func (e *TiFlashSystemTableRetriever) initialize(sctx sessionctx.Context, tiflashInstances set.StringSet) error {
-	store := sctx.GetStore()
-	if etcd, ok := store.(kv.EtcdBackend); ok {
-		var addrs []string
-		var err error
-		if addrs, err = etcd.EtcdAddrs(); err != nil {
+	storeInfo, err := infoschema.GetStoreServerInfo(sctx)
+	if err != nil {
+		return err
+	}
+
+	for _, info := range storeInfo {
+		if info.ServerType != kv.TiFlash.Name() {
+			continue
+		}
+		info.ResolveLoopBackAddr()
+		if len(tiflashInstances) > 0 && !tiflashInstances.Exist(info.Address) {
+			continue
+		}
+		hostAndStatusPort := strings.Split(info.StatusAddr, ":")
+		if len(hostAndStatusPort) != 2 {
+			return errors.Errorf("node status addr: %s format illegal", info.StatusAddr)
+		}
+		// fetch tiflash config
+		configURL := fmt.Sprintf("%s://%s/config", util.InternalHTTPSchema(), info.StatusAddr)
+		req, err := http.NewRequest(http.MethodGet, configURL, nil)
+		if err != nil {
 			return err
 		}
-		if addrs != nil {
-			domainFromCtx := domain.GetDomain(sctx)
-			if domainFromCtx != nil {
-				cli := domainFromCtx.GetEtcdClient()
-				prefix := "/tiflash/cluster/http_port/"
-				kv := clientv3.NewKV(cli)
-				ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-				resp, err := kv.Get(ctx, prefix, clientv3.WithPrefix())
-				cancel()
-				if err != nil {
-					return errors.Trace(err)
-				}
-				for _, ev := range resp.Kvs {
-					id := string(ev.Key)[len(prefix):]
-					if len(tiflashInstances) > 0 && !tiflashInstances.Exist(id) {
-						continue
-					}
-					url := fmt.Sprintf("%s://%s", util.InternalHTTPSchema(), ev.Value)
-					req, err := http.NewRequest(http.MethodGet, url, nil)
-					if err != nil {
-						return errors.Trace(err)
-					}
-					resp, err := util.InternalHTTPClient().Do(req)
-					if err != nil {
-						sctx.GetSessionVars().StmtCtx.AppendWarning(err)
-						continue
-					}
-					resp.Body.Close()
-					e.instanceInfos = append(e.instanceInfos, tiflashInstanceInfo{
-						id:  id,
-						url: url,
-					})
-					e.instanceCount += 1
-				}
-				e.initialized = true
-				return nil
+		resp, err := util.InternalHTTPClient().Do(req)
```

Comment:
Maybe we can use `util.InternalHTTPClient().Get(configURL)` for convenience?

Reply:
Fixed.